src/components/TimeoutModal/TimeoutModal.test.js

```diff
@@ -0,0 +1,119 @@
+            setModalTimeout: setModalTimeoutMock,
+        });
+        delete window.location;
+        window.location = {
+            href: '',
+            assign: jest.fn(),
+        }
+    });
+    afterEach(() => {
+        jest.useRealTimers();
+        jest.clearAllMocks();
+    });
+
+    test('shows modal after 90% of session time has elapsed', () => {
+        render(<TimeoutModal />);
+        // Simulate the passage of time to open the modal
+        act(() => {
+            jest.advanceTimersByTime(9000);
+        });
+        waitFor(() => expect(screen.getByTestId('timeout-modal')).toBeVisible());
+    });
+
+    test('extends session when Extend button is clicked', () => {
+        axios.get.mockResolvedValue({
+            data: {
+                result: {
+                    logged_in: true,
+                    session_timeout: 10000,
+                    user: {
+                        first_name: 'John',
+                        last_initial: 'D',
+                        user_id: '123',
+                        has_profile: true,
+                        tenant: 'tenant',
+                        has_applications: true,
+                        roles: ['role1'],
+                    },
+                    itrust_idp: 'itrust_idp',
+                    itrust_url: 'itrust_url',
+                    okta_idp: 'okta_idp',
+                    okta_login_and_redirect_url: 'okta_login_and_redirect_url',
+                    is_chair: false,
+                    is_manager: false,
+                    is_exec_sec: false,
+                },
+            },
+        });
+
+        render(<TimeoutModal />);
+        // Simulate the passage of time to open the modal
+        act(() => {
+            jest.advanceTimersByTime(9000);
+        });
+        expect(screen.getByTestId('timeout-modal')).toBeVisible()
+        fireEvent.click(screen.getByText('Extend'));
+        waitFor(() => expect(setAuthMock).toHaveBeenCalled());
+    });
+
+    test('logs out user when Logout button is clicked', () => {
+        const mockedFunction = jest.fn(() => location.href = '/logout.do');
+        render(<TimeoutModal onClick={mockedFunction()} />);
+        // Simulate the passage of time to open the modal
+        act(() => {
+            jest.advanceTimersByTime(9000);
+        });
+        waitFor(() => expect(screen.getByTestId('timeout-modal')).toBeVisible());
+        fireEvent.click(screen.getByText('Logout'));
+        expect(mockedFunction).toHaveBeenCalledTimes(1);
+        expect(window.location.href).toBe('/logout.do');
+    });
+
+    test('auto closes modal and logs out user after remaining time', async () => {
+        render(<TimeoutModal />);
+
+        // Simulate the passage of time to open the modal
+        act(() => {
+            jest.advanceTimersByTime(9000);
+        });
+
+        // Wait for the modal to be visible
+        await waitFor(() => expect(screen.getByTestId('timeout-modal')).toBeVisible());
```

Comment:
Do we need both await and waitfor here ? This is just an observation, so ignore if it is not relevant.

Reply:
The test I was running was passing, but throwing a warning that because there are two waitfor statements, I should use await on the first one